src/components/Markets/index.tsx

```diff
@@ -123,52 +65,85 @@ export class Markets extends React.Component<MarketsProps, MarketsState> {
+        return isChangeValue ? renderChange(cell as string) : cell;
+    }, [renderChange]);
+
+    const filterType = (headerKey: string, key: string) => (item: CellData[]) => {
+        const typeIndex = props.headers ? props.headers.indexOf(headerKey) : defaultHeaders.indexOf(headerKey);
+
+        return (item[typeIndex] as string).includes(key);
     };
 
-    private filterType = (headerKey: string, searchKey: string) => (item: CellData[]) => {
-        const typeIndex = this.props.headers ? this.props.headers.indexOf(headerKey) : this.defaultHeaders.indexOf(headerKey);
+    const createUniqueCurrencies = (currencies: string[], market: string) => {
+        const isCurrencyUnique = (currency: string) => !currencies.includes(currency);
+
+        const marketCurrencies = market.split('/').map((c: string) => c.trim());
+        const uniqueCurrencies = marketCurrencies.filter(isCurrencyUnique);
 
-        return (item[typeIndex] as string).includes(searchKey);
+        return currencies.concat(uniqueCurrencies);
     };
 
-    private get filters() {
-        const { data } = this.props;
+    const transformCurrencyToFilter = (currency: string) => ({
+        name: currency,
+        filter: filterType('Pair', currency),
+    });
+
+    const filtersFn = () => {
+        const { data } = props;
 
         const currencyFilters = data && data.length > 0
-            ? this.props.data
-                .map(this.getMarketFromDataRow)
-                .reduce(this.createUniqueCurrencies, [])
-                .map(this.transformCurrencyToFilter)
+            ? data
+                .map(getMarketFromDataRow)
+                .reduce(createUniqueCurrencies, [])
+                .map(transformCurrencyToFilter)
             : [];
 
         return [
             {
                 name: 'All',
-                filter: this.filterType('Pair', ''),
+                filter: filterType('Pair', ''),
             },
             ...currencyFilters,
         ];
-    }
-
-    private getMarketFromDataRow = (market: React.ReactNode[]) => market[0] as string;
-
-    private createUniqueCurrencies(currencies: string[], market: string) {
-        const isCurrencyUnique = (currency: string) => !currencies.includes(currency);
-
-        const marketCurrencies = market.split('/').map((c: string) => c.trim());
-        const uniqueCurrencies = marketCurrencies.filter(isCurrencyUnique);
-
-        return currencies.concat(uniqueCurrencies);
-    }
+    };
 
-    private transformCurrencyToFilter = (currency: string) => ({
-        name: currency,
-        filter: this.filterType('Pair', currency),
-    });
-}
+    const getMarketFromDataRow = (market: React.ReactNode[]) => market[0] as string;
+
+    const { filters = true, headers, title, filterPlaceholder = '', rowKeyIndex, selectedKey } = props;
+
+    let tableData = filteredData.length > 0
+        ? filteredData
+        : [['', '', '']];
```

Comment:
Need to change ```[['', '', '']]``` and pass just empty array ```[]``` and change all places like this

Reply:
Not quite sure what you mean. It was like this before my changes